src/tests/class_tests/openms/source/AbsoluteQuantitation_test.cpp

```diff
@@ -0,0 +1,309 @@
+  AbsoluteQuantitation absquant;
+  String feature_name = "peak_apex_int";
+  double inf = 1.0/0.0;
+  // dummy features
+  OpenMS::Feature component_1, component_2;
+  component_1.setMetaValue(feature_name, 5.0);
+  component_1.setMetaValue("native_id","component1");
+  component_2.setMetaValue(feature_name, 5.0);
+  component_2.setMetaValue("native_id","component2");
+  // tests
+  TEST_REAL_SIMILAR(absquant.calculateRatio(component_1,component_2,feature_name),1.0);
+  component_2.setMetaValue(feature_name, 0.0);
+  TEST_REAL_SIMILAR(absquant.calculateRatio(component_1,component_2,feature_name),inf);
+  // dummy features
+  OpenMS::Feature component_3, component_4;
+  component_3.setMetaValue("peak_area", 5.0);
+  component_3.setMetaValue("native_id","component3");
+  component_4.setMetaValue("peak_area", 5.0);
+  component_4.setMetaValue("native_id","component4");
+  TEST_REAL_SIMILAR(absquant.calculateRatio(component_1,component_4,feature_name),5.0);
+  TEST_REAL_SIMILAR(absquant.calculateRatio(component_3,component_4,feature_name),0.0);
+END_SECTION
+
+START_SECTION((double calculateBias(const double & actual_concentration, const double & calculated_concentration)))
+  AbsoluteQuantitation absquant;
+  double actual_concentration = 5.0;
+  double calculated_concentration = 5.0;
+  TEST_REAL_SIMILAR(absquant.calculateBias(actual_concentration,calculated_concentration),0.0);
+  calculated_concentration = 4.0;
+  TEST_REAL_SIMILAR(absquant.calculateBias(actual_concentration,calculated_concentration),20.0);
+END_SECTION
+
+START_SECTION((double applyCalibration(const Feature & component,
+  const Feature & IS_component,
+  const String & feature_name,
+  const String & transformation_model,
+  const Param & transformation_model_params)))
+
+  AbsoluteQuantitation absquant;
+
+  // set-up the features
+  Feature component, IS_component;
+  component.setMetaValue("native_id","component");
+  component.setMetaValue("peak_apex_int",2.0);
+  IS_component.setMetaValue("native_id","IS");
+  IS_component.setMetaValue("peak_apex_int",1.0);
+  String feature_name = "peak_apex_int";
+
+  // set-up the model and params
+  // y = m*x + b
+  // x = (y - b)/m
+  String transformation_model;
+  Param param;
+  transformation_model = "TransformationModelLinear";  
+  param.setValue("slope",2.0);
+  param.setValue("intercept",1.0);
+
+  TEST_REAL_SIMILAR(absquant.applyCalibration(component,
+    IS_component,
+    feature_name,
+    transformation_model,
+    param),0.5);
+END_SECTION
+
+START_SECTION((void quantifyComponents(std::vector<FeatureMap>& unknowns)))
+
+  AbsoluteQuantitation absquant;
+
+  // set-up the unknown FeatureMap
+  FeatureMap unknown_feature_map;
+  // set-up the features and sub-features
+  std::vector<Feature> unknown_feature_subordinates;
+  Feature unknown_feature, component, IS_component;
+  String feature_name = "peak_apex_int";
+  // component 1
+  unknown_feature.setMetaValue("PeptideRef","component_group1");
```

Comment:
again, this works for now but definitely is not the way to go forward. One problem is that we also use PeptideRef in the TOPPView GUI app to group together chromatograms.

Reply:
I am up for suggestions on this.

Reply:
this is not an issue with your PR but with our way of storing component ids in features - its pretty ugly :-(

Reply:
I agree.